app/models/partner.rb

```diff
@@ -13,7 +13,25 @@ def ensure_role
     self.role = 'partner'
   end
 
+  def balance
+    wallet.contribution_balance.amount_bat + invoice_amount
+  end
+
+  def balance_in_currency
+    wallet.contribution_balance.add_bat(invoice_amount)
+    wallet.contribution_balance.amount_default_currency
+  end
+
   def name
     self[:name] || self[:email]
   end
+
+  private
+
+  def invoice_amount
```

Comment:
@yachtcaptain23 passed me this feedback the other day, I think it applies to this section 

consider 

```ruby 
def invoice_amount
    invoices = Invoice.where(partner_id: id, paid: false)
    amounts = invoices.map { |i| i.finalized_amount || i.amount }

    amounts.map { |x| x.tr(",", "").to_i }.reduce(:+) || 0
end
```

His rationale is prevent `add_bat` from throwing an error if the amount is `nil`

Reply:
Done! 
https://github.com/brave-intl/publishers/pull/1667/commits/91b10f64f07745e510dcba4d2625bb439f90a752